plugins/jobs/web_client/models/JobModel.js

```diff
@@ -36,8 +36,9 @@ var JobModel = AccessControlledModel.extend({
             }));
         }
 
-        this.set('segments', segments);
```

Comment:
This seems to change the behavior of this function. Is there a good reason we shouldn't call this `set()`?